### PR TITLE
Add type parameter for Id in createEntityAdapter

### PR DESCRIPTION
**Pull request by Matt-Ord**
Fixes issue #2098
Let me know if this change also needs any docs/ I've messed up the system for creating a PR

### CODE REVIEW

**Comment by phryneas** [comment]
I've gone over this - from my side it's okay @markerikson . 
I've also roughly gone over https://github.com/reduxjs/redux-toolkit/pull/948 in the context of this - it doesn't seem like the types changes would overlap too much, probably a few types would get one new type argument from each PR, but most would be affected by only one of the two PRs.

**Review by phryneas — APPROVED** [review]

**Comment by Matt-Ord** [comment]
Do I need to do anything to update the docs- I think currently they have still have the old types

File: packages/toolkit/src/entities/models.ts [hunk 1]
```diff
@@ -24,7 +24,7 @@ export type Dictionary<T, Id extends EntityId> = Partial<Record<Id, T>>
 /**
  * @public
  */
-export type DictionaryNum<T> = Dictionary<TemplateStringsArray, number>
```
Comment (phryneas) [inline packages/toolkit/src/entities/models.ts]: I think that with this PR, this type could be pretty much completely removed - or only be kept for backwards compat, but not really referenced any more/

**Comment by phryneas** [comment]
Yeah, docs PRs would be great where you see them necessary 👍

**Comment by Matt-Ord** [comment]
I think I've updated the docs where it's needed - it should be ready to go

**Comment by markerikson** [comment]
Hiya. I'm looking at this, and it seems like it makes some decidedly breaking changes to the types: it looks like we _have_ to pass `, Id` everywhere (including existing public types), and it also rewrites the `Dictionary` type.

Given that, I don't think we can ship this in a 1.x release, unless I'm missing something.

**Comment by Matt-Ord** [comment]
I would agree this is probably a breaking change - although I think it will be 'worth it' when you come to a 2.0 release, since it's solved a real problem on a medium to large codebase I've been working on.

Ideally I would say swap the order of Id and T (and do the same on all types) but this would make it 'more breaking'  - is this the issue you have with the dictionary type?

**Comment by Matt-Ord** [comment]
Just seen this pr was closed - is this not something you want to change anymore? I'm happy to put the work in to update it if we're coming up to a 2.0 release?

**Comment by phryneas** [comment]
I think this was just an accident.

**Comment by markerikson** [comment]
Uh. Yeah, _I_ don't have permission to reopen it either! I think the issue is it was targeted to the `v1.9` branch.

**Pull request closed without merging**